checker/src/lib.rs

```diff
@@ -1,14 +1,25 @@
 #![doc = include_str!("../README.md")]
+#![deny(clippy::all)]
+#![deny(clippy::pedantic)]
 #![allow(
 	unreachable_code,
 	unused_variables,
 	unused_imports,
 	unused_mut,
 	dead_code,
 	irrefutable_let_patterns,
-	deprecated
+	deprecated,
+    // TODO: Remove when fixed
+	clippy::result_unit_err,
+    clippy::default_trait_access,
+    clippy::missing_errors_doc,
+    clippy::missing_panics_doc,
+    clippy::implicit_hasher,
+    clippy::too_many_lines,
+    // More explicit sometimes to have the module name
+    clippy::module_name_repetitions
 )]
-
+#![warn(clippy::cast_precision_loss, clippy::cast_possible_truncation, clippy::cast_sign_loss)]
```

Comment:
Denying casting can be useful if we want to make sure numbers are well handled, but imo only having it as a warning for now is nice to be aware of it